pybio/spec/utils/build_spec.py

```diff
@@ -0,0 +1,179 @@
+import os
+import datetime
+import hashlib
+import numpy as np
+
+import pybio.spec.raw_nodes as raw_nodes
+
+#
+# utility functions to build the spec from python
+#
+
+
+# TODO refactor common stuff here
+def _build_spec():
+    pass
+
+
+# TODO this should handle both local file paths and urls and download the url somwhere temp
```

Comment:
shouldn't it leave urls as is if this is going to become a model.yaml?

Reply:
It needs to download the urls in order to compute the hashes etc. The urls that are written to the model are indeed not changed.